pkg/util/workloadspread/workloadspread.go

```diff
@@ -61,3 +130,397 @@ func VerifyGroupKind(ref *appsv1alpha1.TargetReference, expectedKind string, exp
+			break
+		}
+	}
+	if subset == nil {
+		return false, nil
+	}
+
+	if pod.Labels == nil {
+		pod.Labels = map[string]string{}
+	}
+	if pod.Annotations == nil {
+		pod.Annotations = map[string]string{}
+	}
+	// inject toleration
+	if len(subset.Tolerations) > 0 {
+		pod.Spec.Tolerations = append(pod.Spec.Tolerations, subset.Tolerations...)
+	}
+	if pod.Spec.Affinity == nil {
+		pod.Spec.Affinity = &corev1.Affinity{}
+	}
+	if pod.Spec.Affinity.NodeAffinity == nil {
+		pod.Spec.Affinity.NodeAffinity = &corev1.NodeAffinity{}
+	}
+	if len(subset.PreferredNodeSelectorTerms) > 0 {
+		pod.Spec.Affinity.NodeAffinity.PreferredDuringSchedulingIgnoredDuringExecution = append(pod.Spec.Affinity.NodeAffinity.PreferredDuringSchedulingIgnoredDuringExecution,
+			subset.PreferredNodeSelectorTerms...)
+	}
+	if subset.RequiredNodeSelectorTerm != nil {
+		if pod.Spec.Affinity.NodeAffinity.RequiredDuringSchedulingIgnoredDuringExecution == nil {
+			pod.Spec.Affinity.NodeAffinity.RequiredDuringSchedulingIgnoredDuringExecution = &corev1.NodeSelector{}
+		}
+		if len(pod.Spec.Affinity.NodeAffinity.RequiredDuringSchedulingIgnoredDuringExecution.NodeSelectorTerms) == 0 {
+			pod.Spec.Affinity.NodeAffinity.RequiredDuringSchedulingIgnoredDuringExecution.NodeSelectorTerms = []corev1.NodeSelectorTerm{
+				*subset.RequiredNodeSelectorTerm,
+			}
+		} else {
+			for i := range pod.Spec.Affinity.NodeAffinity.RequiredDuringSchedulingIgnoredDuringExecution.NodeSelectorTerms {
+				selectorTerm := &pod.Spec.Affinity.NodeAffinity.RequiredDuringSchedulingIgnoredDuringExecution.NodeSelectorTerms[i]
+				selectorTerm.MatchExpressions = append(selectorTerm.MatchExpressions, subset.RequiredNodeSelectorTerm.MatchExpressions...)
+				selectorTerm.MatchFields = append(selectorTerm.MatchFields, subset.RequiredNodeSelectorTerm.MatchFields...)
+			}
+		}
+	}
+	if subset.Patch.Raw != nil {
+		cloneBytes, _ := json.Marshal(pod)
+		modified, err := strategicpatch.StrategicMergePatch(cloneBytes, subset.Patch.Raw, &corev1.Pod{})
+		if err != nil {
+			klog.Errorf("failed to merge patch raw %s", subset.Patch.Raw)
+			return false, err
+		}
+		newPod := &corev1.Pod{}
+		if err = json.Unmarshal(modified, newPod); err != nil {
+			klog.Errorf("failed to unmarshal %s to Pod", modified)
+			return false, err
+		}
+		*pod = *newPod
+	}
+
+	injectWS := &InjectWorkloadSpread{
+		Name:   ws.Name,
+		Subset: subsetName,
+		UID:    generatedUID,
+	}
+	by, _ := json.Marshal(injectWS)
+	pod.Annotations[MatchedWorkloadSpreadSubsetAnnotations] = string(by)
+	return true, nil
+}
+
+func getSpecificSubset(ws *appsv1alpha1.WorkloadSpread, specifySubset string) *appsv1alpha1.WorkloadSpreadSubsetStatus {
+	for _, subset := range ws.Status.SubsetStatuses {
+		if specifySubset == subset.Name {
+			return &subset
+		}
+	}
+	return nil
+}
+
+func (h *Handler) getSuitableSubset(ws *appsv1alpha1.WorkloadSpread) (*appsv1alpha1.WorkloadSpreadSubsetStatus, error) {
+	for i := range ws.Status.SubsetStatuses {
```

Comment:
hits function does not return any error, consider removing the second return value